### PR TITLE
[Fix]: URL isn't fully visible in the tooltip

### PR DESCRIPTION
**Pull request by fedirjh**
### Details

- https://github.com/Expensify/App/pull/10504 caused an issue in Desktop / Web preventing long URLs from breaking text to new line
- Fixed wrong width calculation that leads to This issue : https://github.com/Expensify/App/issues/9331

### Fixed Issues

$ https://github.com/Expensify/App/issues/11486   
PROPOSAL: https://github.com/Expensify/App/issues/11486#issuecomment-1287358882

### test 1 : Long URL

1. Navigate to any chat
2. Send a long URL : select from list 
-   [Url 1](https://www.expensify.com/inbox?policyID=D189E4B5582EAE8D&taskID=AddWorkEmail&policyID=D189E4B5582EAE8D&taskID=AddWorkEmailpolicyID=D189E4B5582EAE8D&taskID=AddWorkEmail&policyID=D189E4B5582EAE8D&taskID=AddWorkEmailpolicyID=D189E4B5582EAE8D&taskID=AddWorkEmail&policyID=D189E4B5582EAE8D&taskID=AddWorkEmail&policyID=D189E4B5582EAE8D&taskID=AddWorkEmail)
- [Url 2](https://www.expensify.com/inbox?policyID=D189E4B5582EAE8DtaskID=AddWorkEmailpolicyID=D189E4B5582EAE8D&taskID=AddWorkEmailpolicyID=D189E4B5582EAE8DtaskID=AddWorkEmailpolicyID=D189E4B5582EAE8DtaskID=AddWorkEmailpolicyID=D189E4B5582EAE8DtaskID=AddWorkEmailpolicyID=D189E4B5582EAE8DtaskID=AddWorkEmailpolicyID=D189E4B5582EAE8DtaskID=AddWorkEmail)
- [Url 3](https://www.expensify.com/inbox?policyIDD189E4B5582EAE8DtaskIDAddWorkEmailpolicyIDD189E4B5582EAE8D&taskIDAddWorkEmailpolicyIDD189E4B5582EAE8DtaskIDAddWorkEmailpolicyID=D189E4B5582EAE8DtaskIDAddWorkEmailpolicyID189E4B5582EAE8DtaskIDAddWorkEmailpolicyIDD189E4B5582EAE8DtaskIDAddWorkEmailpolicyIDD189E4B5582EAE8DtaskIDAddWorkEmail)
4. Hover
5. The whole Url is visible

### Test 2 : Pin / Upin

1. Open any chat
2. Hover over "Pin" icon and press it.
3. The Unpin tooltip should be in one line without being trimmed/wrapped

### Test 3 : Copy Email/Phone no.

1. Open Desktop Website/App
2. Navigate to Chat > Details
3. Copy Email/Phone no.
4. The Copy to clipboard text should be in one line without being trimmed/wrapped

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web

Web - Safari
https://user-images.githubusercontent.com/36869046/199615934-27c3d7c3-7147-4ee5-be6b-100e975f6e30.mov

Web - Chrome
https://user-images.githubusercontent.com/36869046/199616069-6d1b869b-935c-4e37-9560-7836a98c2855.mov

Web - Firefox
https://user-images.githubusercontent.com/36869046/199616362-a9760aa6-7c18-4212-8671-0cdb8264e4c1.mov

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/36869046/199736233-38d1e522-d980-4b33-9e4b-dc1e210a9408.webm

#### Mobile Web - Safari

https://user-images.githubusercontent.com/36869046/199736387-0c100a34-56e1-494c-9b29-d52864c70d36.mp4

#### Desktop

https://user-images.githubusercontent.com/36869046/199616248-cd92d4b6-fd05-4a1a-b87c-b6e0e5065b58.mov

#### iOS

https://user-images.githubusercontent.com/36869046/199735809-3afe9880-d2fb-4998-bf0f-133eafad970a.mov

#### Android

https://user-images.githubusercontent.com/36869046/199735974-42016d34-4752-46b5-be6b-bb456e04c34b.mov

### CODE REVIEW

**Review by sobitneupane — COMMENTED** [review]
@fedirjh Please Add screenshots for native and mWeb as well. We just want to make sure that the changes made here do not crash them.

Please add test steps for tooltips of Pin/Unpin and Copy to clipboard/copied as well.

**Comment by fedirjh** [comment]
> @fedirjh Please Add screenshots for native and mWeb as well. We just want to make sure that the changes made here do not crash them.
> 
> Please add test steps for tooltips of Pin/Unpin and Copy to clipboard/copied as well.

@sobitneupane Tooltip doesn't work on mobile or native  
https://github.com/Expensify/App/pull/10504#issuecomment-1227754356

**Comment by sobitneupane** [comment]
> Tooltip doesn't work on mobile or native

@fedirjh Yes, You are right. We just want to make sure that changes do not crash native app/ mWeb.

> We just want to make sure that the changes made here do not crash them.

**Comment by fedirjh** [comment]
@sobitneupane here are tests

### iOS

https://user-images.githubusercontent.com/36869046/199735809-3afe9880-d2fb-4998-bf0f-133eafad970a.mov

### Android

https://user-images.githubusercontent.com/36869046/199735974-42016d34-4752-46b5-be6b-bb456e04c34b.mov

### mWeb Chrome

[mWeb Chrome.webm](https://user-images.githubusercontent.com/36869046/199736233-38d1e522-d980-4b33-9e4b-dc1e210a9408.webm)

### mWeb Safari

https://user-images.githubusercontent.com/36869046/199736387-0c100a34-56e1-494c-9b29-d52864c70d36.mp4

**Comment by fedirjh** [comment]
@sobitneupane can you request review from @thienlnam too ?

**Comment by sobitneupane** [comment]
@fedirjh Can you please add those native and mWeb [Screenshots](https://github.com/Expensify/App/pull/12408#issuecomment-1302135441) in [PR body](https://github.com/Expensify/App/pull/12408#issue-1433884735).

Also you might have missed the below comment. Please update test steps in PR body to address the following.
>Please add test steps for tooltips of Pin/Unpin and Copy to clipboard/copied as well.


> @sobitneupane can you request review from @thienlnam too ?

I cannot. Once done with my review. I will let him know.

**Comment by fedirjh** [comment]
> @fedirjh Can you please add those native and mWeb [Screenshots](https://github.com/Expensify/App/pull/12408#issuecomment-1302135441) in [PR body](https://github.com/Expensify/App/pull/12408#issue-1433884735).

> Please add test steps for tooltips of Pin/Unpin and Copy to clipboard/copied as well.

@sobitneupane done .

**Review by sobitneupane — APPROVED** [review]
@thienlnam Changes look good and test well on all platforms.
🎀👀🎀 C+ reviewed

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots


#### Web


https://user-images.githubusercontent.com/25876548/200032424-0aa3f697-000d-4056-9760-64784f7e3de4.mov


#### Mobile Web - Chrome



#### Mobile Web - Safari




#### Desktop


https://user-images.githubusercontent.com/25876548/200027674-a8381970-f289-457d-a11e-b71ee67b2368.mov


#### iOS



#### Android

File: src/components/Tooltip/TooltipRenderedOnPageBody.js [hunk 1]
```diff
@@ -77,7 +77,10 @@ class TooltipRenderedOnPageBody extends React.PureComponent {
         if (prevProps.text === this.props.text) {
             return;
         }
-        this.updateTooltipTextWidth();
+
+        // Reset the tooltip text width to 0 so that we can measure it again.
+        // eslint-disable-next-line react/no-did-update-set-state
+        this.setState({tooltipTextWidth: 0}, this.updateTooltipTextWidth);
```
Comment (thienlnam) [inline src/components/Tooltip/TooltipRenderedOnPageBody.js:83]: This lint is correct - the placement of this is not ideal as we're adding infinite re-renders here
Can you find a better placement to reset the tooltip text width?

**Review by thienlnam — CHANGES_REQUESTED** [review]

File: src/components/Tooltip/TooltipRenderedOnPageBody.js (hunk 1 above)
Comment (fedirjh) [reply src/components/Tooltip/TooltipRenderedOnPageBody.js:83]:
```jsx
      if (prevProps.text === this.props.text) {
            return;
        }
```
this line will prevent the infinite re-rendering , we are only updating tooltipTextWidth , once , when text changes

Comment (thienlnam) [reply src/components/Tooltip/TooltipRenderedOnPageBody.js:83]: ah good point, missed that

**Review by thienlnam — APPROVED** [review]

**Comment by mountiny** [comment]
Just testing something here.

**Pull request merged**